File: src/nodes/bad.cpp
#include "nodes/bad.hpp"
#include <iostream>

using namespace nodes;

NodeBad::NodeBad(const utils::Position &position)
    : Node(position) {

}

void NodeBad::show(std::string &indent, char end) const {
    std::cout << indent << "NodeBad" << end;
}

NodeKind NodeBad::getKind() const {
    return NodeKind::Bad;
}

void NodeBad::emit(core::Emitter &) const {
    
}

void NodeBad::setId() {
    
}

utils::OptimizedResult NodeBad::optimize(core::Optimizer &) {
    return utils::OptimizedResult(utils::OptimizedFlag::None);
}

File: src/nodes/bad.rs
use crate::utils::position::Position;

#[derive(Clone)]
pub struct NodeBad {
    position: Position,
}

impl NodeBad {
    pub fn init(position: Position) -> NodeBad {
        NodeBad { position }
    }
    
    pub fn get_position(&self) -> &Position {
        &self.position
    }
    
    pub fn show(&self, indent: &mut String, end: char) {
        print!("{indent}NodeBad{end}");
    }
}